mig/shared/safeinput.py

```diff
@@ -884,8 +884,14 @@ def valid_email_address(addr, allow_real_name=True):
         raise InputException("Only plain addresses without name allowed")
     if not silent_email_validator(plain_addr):
         raise InputException("No actual email address included")
-    merged = formataddr(name_address_pair)
-    if merged != addr:
+    raise_error = False
+    try:
+        merged = formataddr(name_address_pair)
+        if merged != addr:
+            raise InputException("")
+    except (InputException, UnicodeEncodeError) as exc:
```

Comment:
Hmmm, perhaps I need to understand the exact mechanics here, but I don't have a great gut-feeling about ever catching and bailing on `XEncodeError`/`DecodeError`s - in the sense that I think it indicates that we may be covering another fundamental issue or at least missed an explicit `force_X`-wrap in that case.
Did you have a particular code path ending up there?

Reply:
IIRC the situation is you can end up with both these cases when this code runs on Python 3 - there is "our" code that triggers InputException (and I use this to our advantage to force an error on line 891) but you have the addition problem that on Py3 formataddr can blow up in just its string handling due to getting unexpected unicode chars. The latter are forbidden by our checks anyway, so it seemed prudent to just handle it here - note that iirc the tests are pretty exhaustive and they are working across both versions.